MarsRover/Rover.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace MarsRover
+{
+    public class Rover
+    {
+        private Grid _grid;
+        private IHeading _heading;
+        private IEnumerable<Obstacle> _obstacles;
+        private bool _obstacleDetected;
+        private int _obstacleDetectedCoordinateX;
+        private int _obstacleDetectedCoordinateY;
+
+        public Rover(Grid grid, IHeading heading, IEnumerable<Obstacle> Obstacles)
+        {
+            _grid = grid;
+            _heading = heading;
+            _obstacles = Obstacles;
+            _obstacleDetected = false;
+        }
+
+        public void TurnLeft() => _heading = _heading.TurnLeft();
+
+        public void TurnRight() => _heading = _heading.TurnRight();
+
+        public void MoveForward() => _grid = _heading.MoveForward(_grid);
+
+        public void MoveBackward() => _grid = _heading.MoveBackward(_grid);
+
+        public void InvalidCommand() { }
+
+        public string Direction() => _heading.GetHeading();
+
+        public int XCoordinate() => _grid.XCoordinate();
+
+        public int YCoodinate() => _grid.YCoordinate();
+
+        public bool ObstacleDetected() => _obstacleDetected;
+
+        public int ObstacleDetectedXCoordinate() => _obstacleDetectedCoordinateX;
+
+        public int ObstacleDetectedYCoordinate() => _obstacleDetectedCoordinateY;
+
+        public Rover ExecuteCommands(string commands)
+        {
+            foreach (var letter in commands)
+            {
+                var commandInput = new InputCommands(letter.ToString());
+                var commandFactory = commandInput.ReturnCommand();
+                var command = commandFactory(this);
+                var previousLocation = _grid;
```

Comment:
Do we still need to store the `previousLocation` since we got rid of the history?

Reply:
Yes, we still need that.  The previousLocation grid is what replaced the unnecessary tracking of all the rover movements being collected in the old history procedure.  This allows for making sure the rover coordinates are set to the spot before an obstacle was detected.